api/types/server.go

```diff
@@ -392,12 +394,39 @@ func (s *ServerV2) CheckAndSetDefaults() error {
 	// defaults.ServerAnnounceTTL).
 	s.setStaticFields()
 	if err := s.Metadata.CheckAndSetDefaults(); err != nil {
-		return trace.Wrap(err)
+		// if the server is a registered OpenSSH node, allow the name
+		// to be randomly generated
+		if s.SubKind == KindOpenSSHNode && trace.UserMessage(err) == "missing parameter Name" {
```

Comment:
Instead of relying on returned error, can we do explicit check `if s.SubKind == KindOpenSSHNode && s.Metadata.Name == "" {...}` before metadata's CheckAndSetDefaults call above?

Reply:
:facepalm: that's a much better way to handle it, idk why I didn't think of that originally, thanks. Done.